### ISSUE TITLE
Error running simple gadgetfinder

### ISSUE THREAD

**Issue by Hooshiman** [angr/angrop#121]
### Description

I run the code as it is said in the readme and I get this.

Traceback (most recent call last):
  File "/home/kali/tdde63/myenv/lib/python3.11/site-packages/angrop/gadget_finder/gadget_analyzer.py", line 65, in analyze_gadget
    ctrl_type = self._check_for_control_type(init_state, final_state)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kali/tdde63/myenv/lib/python3.11/site-packages/angrop/gadget_finder/gadget_analyzer.py", line 454, in _check_for_control_type
    if self._check_if_stack_controls_ast(ip, init_state):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kali/tdde63/myenv/lib/python3.11/site-packages/angrop/gadget_finder/gadget_analyzer.py", line 544, in _check_if_stack_controls_ast
    rop_utils.fast_unconstrained_check(initial_state, ast)
  File "/home/kali/tdde63/myenv/lib/python3.11/site-packages/angrop/rop_utils.py", line 114, in fast_unconstrained_check
    for a in ast.recursive_children_asts:
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'BV' object has no attribute 'recursive_children_asts'

### Steps to reproduce the bug

import angr, angrop
p = angr.Project("/bin/ls")
rop = p.analyses.ROP()
rop.find_gadgets()

### Environment

angr environment report
=============================
Date: 2024-10-02 05:23:27.356617
Running in virtual environment at /home/kali/tdde63/myenv
Platform: linux-x86_64
Python version: 3.11.9 (main, Apr 10 2024, 13:16:36) [GCC 13.2.0]
######## angr #########
Python found it in /home/kali/tdde63/myenv/lib/python3.11/site-packages/angr/__init__.py
Pip version 9.2.120
Couldn't find git info
######## ailment #########
Python found it in /home/kali/tdde63/myenv/lib/python3.11/site-packages/ailment/__init__.py
Pip version 9.2.120
Couldn't find git info
######## cle #########
Python found it in /home/kali/tdde63/myenv/lib/python3.11/site-packages/cle/__init__.py
Pip version 9.2.120
Couldn't find git info
######## pyvex #########
Python found it in /home/kali/tdde63/myenv/lib/python3.11/site-packages/pyvex/__init__.py
Pip version 9.2.120
Couldn't find git info
######## claripy #########
Python found it in /home/kali/tdde63/myenv/lib/python3.11/site-packages/claripy/__init__.py
Pip version 9.2.120
Couldn't find git info
######## archinfo #########
Python found it in /home/kali/tdde63/myenv/lib/python3.11/site-packages/archinfo/__init__.py
Pip version 9.2.120
Couldn't find git info
######## z3 #########
Python found it in /home/kali/tdde63/myenv/lib/python3.11/site-packages/z3/__init__.py
Pip version 4.13.0.0
Couldn't find git info
######## unicorn #########
Python found it in /home/kali/tdde63/myenv/lib/python3.11/site-packages/unicorn/__init__.py
Pip version 2.0.1.post1
Couldn't find git info
######### Native Module Info ##########
angr: <CDLL '/home/kali/tdde63/myenv/lib/python3.11/site-packages/angr/lib/angr_native.so', handle 3a14b80 at 0x7f95a7413bd0>
unicorn: <CDLL '/home/kali/tdde63/myenv/lib/python3.11/site-packages/unicorn/lib/libunicorn.so.2', handle 3368570 at 0x7f95acf928d0>
pyvex: <cffi.api._make_ffi_library.<locals>.FFILibrary object at 0x7f95ab238350>
z3: <CDLL '/home/kali/tdde63/myenv/lib/python3.11/site-packages/z3/lib/libz3.so', handle 2f11bd0 at 0x7f95ad21fe10>

**Comment by twizmwazin** [comment]
This one is my fault, I’ll have a patch for you to test shortly

**Comment by twizmwazin** [comment]
Actually this has been fixed already on master. I assume you installed via pypi and not git? I’ll post a release today.

**Comment by Hooshiman** [comment]
Okey, Nice to know. I'll try from github tomorrow!

Skickat från Outlook för Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by Hooshiman** [comment]
@twizmwazin I have now downloaded from git and I don't get the same error (might be because I don't even reach it I'm not sure) 
WARNING  | 2024-10-03 03:26:57,799 | angr.state_plugins.unicorn_engine | Unicorn is not installed. Support disabled.                                      
WARNING  | 2024-10-03 03:26:57,804 | angr.state_plugins.unicorn_engine | failed loading "angr_native.so", unicorn support disabled ('NoneType' object has no attribute 'unicorn')                                                      
>>> import angrop
>>> p = angr.Project("/bin/ls")
>>> rop = p.analyses.ROP()
Traceback (most recent call last):
  File "/home/kali/tdde63/myenv/lib/python3.12/site-packages/angr/misc/plugins.py", line 252, in request_plugin
    return self._default_plugins[name]
           ~~~~~~~~~~~~~~~~~~~~~^^^^^^
KeyError: 'ROP'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/kali/tdde63/myenv/lib/python3.12/site-packages/angr/misc/plugins.py", line 77, in __getattr__
    return self.get_plugin(name)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/kali/tdde63/myenv/lib/python3.12/site-packages/angr/misc/plugins.py", line 161, in get_plugin
    plugin_cls: type[P] = self._active_preset.request_plugin(name)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kali/tdde63/myenv/lib/python3.12/site-packages/angr/misc/plugins.py", line 254, in request_plugin
    raise AngrNoPluginError(f"There is no plugin named {name}") from err
angr.errors.AngrNoPluginError: There is no plugin named ROP

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/kali/tdde63/myenv/lib/python3.12/site-packages/angr/misc/plugins.py", line 79, in __getattr__
    raise AttributeError(name) from err
AttributeError: ROP.

I've tried downloading everything in angr from git but when I get to angr it complains about the versions about the other dependencies so in the end I'm just using the pypy version of angr and down. I'm aware this isn't really the angrop but I'm hoping you have some tip. Im glad for all the help I can get.

**Comment by twizmwazin** [comment]
When installing angr from git we have a bunch of repositories that also need to be installed over git. We use the angr-dev repo for that which has some scripts for making this fairly simple. But I'll try to get that release out today to PyPI to spare you that pain. :)

**Comment by twizmwazin** [comment]
Just published angrop 9.2.11. Give it a try and let me know how it goes.
